packit_service/worker/build/build_helper.py

```diff
@@ -65,33 +56,23 @@ def __init__(
         config: ServiceConfig,
         package_config: PackageConfig,
         project: GitProject,
-        event: Union[
-            PullRequestGithubEvent,
-            PullRequestPagureEvent,
-            PullRequestCommentGithubEvent,
-            PullRequestCommentPagureEvent,
-            CoprBuildEvent,
-            PushGitHubEvent,
-            ReleaseEvent,
-            KojiBuildEvent,
-        ],
+        event: dict,
```

Comment:
Cannot we have some class representation for this?

With general dictionaries we are losing the benefit of static typing checks.

Reply:
So would be creating a class which would represent data needed by helpers a better option? Something like 
```python
class BuildHelperData:
    def __init__(
        self,
        event: dict,
    ):
        self.pr_id = event.get("pr_id")
        self.git_ref = event.get("git_ref")
        self.commit_sha = event.get("commit_sha")
        self.trigger = (
            TheJobTriggerType(event.get("trigger")) if event.get("trigger") else None
        )
```

Reply:
Yes if the set of attributes differ to the original event. (Otherwise, it makes sense to reinitialize the original object.)

It allows us to better define and name this context we are forwarding.

Reply:
created